packages/block-editor/src/hooks/font-appearance.js

```diff
@@ -30,53 +34,125 @@ export function FontAppearanceEdit( props ) {
 
 	const fontStyles = useEditorFeature( 'typography.fontStyles' );
 	const fontWeights = useEditorFeature( 'typography.fontWeights' );
-	const isDisabled = useIsFontAppearanceDisabled( props );
+	const isFontStyleDisabled = useIsFontStyleDisabled( props );
+	const isFontWeightDisabled = useIsFontWeightDisabled( props );
 
-	if ( isDisabled ) {
+	if ( isFontStyleDisabled && isFontWeightDisabled ) {
 		return null;
 	}
 
 	const onChange = ( newStyles ) => {
+		// Match style selection with preset and create CSS var style if appropriate.
+		const presetStyle = fontStyles.find(
+			( { slug } ) => slug === newStyles.fontStyle
+		);
+		const newFontStyle = presetStyle
+			? `var:preset|font-style|${ presetStyle.slug }`
+			: undefined;
+
+		// Match weight selection with preset and create CSS var style if appropriate.
+		const presetWeight = fontWeights.find(
+			( { slug } ) => slug === newStyles.fontWeight
+		);
+		const newFontWeight = presetWeight
+			? `var:preset|font-weight|${ presetWeight.slug }`
```

Comment:
Not directly related to this PR but I left a comment regarding the current preset usage on fontStyle and fontWeight PR's.
These things are breaking change and may affect block markup so I guess it is something we should look at and decide as soon as possible
https://github.com/WordPress/gutenberg/pull/26444#pullrequestreview-527630030.

Reply:
This has been updated to have a separate value property in the presets.